staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go

```diff
@@ -161,6 +161,26 @@ type JSONSchemaProps struct {
 	//      Atomic maps will be entirely replaced when updated.
 	// +optional
 	XMapType *string `json:"x-kubernetes-map-type,omitempty" protobuf:"bytes,43,opt,name=xKubernetesMapType"`
+
+	// x-kubernetes-validations describes a list of validation rules for expression validation.
+	// This field is an alpha-level. Using this field requires the feature gate `CustomResourceValidationExpressions` to be enabled.
+	XValidations ValidationRules `json:"x-kubernetes-validations,omitempty" protobuf:"bytes,44,rep,name=xKubernetesValidations"`
+}
+
+// ValidationRules describe a list of validation rules for expression validation
+//  +listType=atomic
+type ValidationRules []ValidationRule
+
+// ValidationRule is used for expression validation.
+type ValidationRule struct {
+	// Rule represents the validation rule which will be evaluated by CEL.
+	// ref: https://github.com/google/cel-spec
+	// The validator will be scoped to the location of the x-kubernetes-validations extension in the schema
+	// and will use `self` to represent the scoped field name.
+	// If an object property collides with a RESERVED keyword, it will be escaped by prepending a _ prefix.
+	Rule string `json:"rule" protobuf:"bytes,1,opt,name=rule"`
+	// Message represents the message displayed when validation failed.
```

Comment:
can we give an example here of the format expected? Is this like "must be X", e.g. "must be a URL with the host matching spec.host" ?